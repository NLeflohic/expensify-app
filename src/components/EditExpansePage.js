import React from "react";

const EditExpensePage = () => <div>this is the edit page</div>;

export default EditExpensePage;
