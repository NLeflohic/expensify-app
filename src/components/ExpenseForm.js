import React from "react";
import moment from "moment";
import "react-dates/initialize";
import { SingleDatePicker } from "react-dates";
import "react-dates/lib/css/_datepicker.css";

moment.locale("fr");
const now = moment();
console.log(now.format("DD MMMM YYYY"));

export default class ExpenseForm extends React.Component {
  state = {
    description: "",
    note: "",
    amount: "",
    createAt: moment(),
    calenderFocused: false
  };
  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = e => {
    const amount = e.target.value;
    if (amount.match(/^\d*(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChanged = () => {
    this.setState(() => ({ createAt }));
  };

  onFocusChanged = ({ focused }) => {
    this.setState(() => ({ calenderFocused: focused }));
  };

  render() {
    return (
      <div>
        <input
          type="text"
          placeholder="Description"
          autoFocus
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input
          type="text"
          placeholder="Amount"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />
        <SingleDatePicker
          date={this.state.createAt} // momentPropTypes.momentObj or null
          onDateChange={this.onDateChanged} // PropTypes.func.isRequired
          focused={this.state.calenderFocused} // PropTypes.bool
          onFocusChange={this.onFocusChanged} // PropTypes.func.isRequired
          id="cal" // PropTypes.string.isRequired,
        />

        <textarea
          placeholder="Add note for your expense (Optionnal)"
          value={this.state.note}
          onChange={this.onNoteChange}
        />
        <button>Add expense</button>
      </div>
    );
  }
}
