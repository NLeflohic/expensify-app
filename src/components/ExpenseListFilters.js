import React from "react";
import { connect } from "react-redux";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate
} from "../actions/filters";

class ExpenseListFilters extends React.Component {
  constructor(props) {
    super(props);
  }

  state = {
    calendarFocused: null
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusedChange = calendarFocused =>
    this.setState(() => ({ calendarFocused }));
  render() {
    return (
      <div>
        <input
          type="text"
          defaultValue={this.props.filters.text}
          onChange={e => {
            props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select
          onChange={e => {
            e.target.value === "date"
              ? props.dispatch(sortByDate())
              : props.dispatch(sortByAmount());
          }}
          value={this.props.filters.sortBy}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
          startDateId="start_dateId" // PropTypes.string.isRequired,
          endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
          endDateId="end_dateId" // PropTypes.string.isRequired,
          onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
          focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
          onFocusChange={this.onFocusedChange} // PropTypes.func.isRequired,
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
