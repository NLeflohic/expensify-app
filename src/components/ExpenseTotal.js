import React from "react";
import { connect } from "react-redux";
import getSumExpenses from "../selectors/expensesTotal";
import selectExpenses from "../selectors/expenses";
import numeral from "numeral";

export const ExpenseTotal = props => {
  return (
    <div>
      <p>
        {props.expensesCount > 0 ? (
          <span>
            Viewing {props.expensesCount} expense(s) totalling{" "}
            {numeral(props.expensesSum / 100).format("$0,00.00")}
          </span>
        ) : (
          <span>No Expenses</span>
        )}
      </p>
    </div>
  );
};

const mapStateToProps = state => ({
  expensesSum: getSumExpenses(state.expenses),
  expensesCount: selectExpenses(state.expenses, state.filters).length
});

export default connect(mapStateToProps)(ExpenseTotal);
