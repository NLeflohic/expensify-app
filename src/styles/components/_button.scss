.button {
  background: $blue;
  padding: $s-size;
  color: white;
  display: inline-block;
  border: none;
  font-size: $font-size-large;
  border-radius: 3px;
  text-decoration: none;
  line-height: 1;
}

.button__link {
  background: none;
}

.button__secondary {
  @extend .button;
  background: #888888;
}
