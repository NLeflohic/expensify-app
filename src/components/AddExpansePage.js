import React from "react";
import { connect } from "react-redux";
import { addExpense } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";

export class AddExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        this is the add page
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addExpense: expense => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
