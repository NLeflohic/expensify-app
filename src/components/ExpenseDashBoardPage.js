import React from "react";
import ExpenseTotal from "./ExpenseTotal";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpenseDashBoardPage = () => (
  <div>
    <ExpenseTotal />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashBoardPage;
