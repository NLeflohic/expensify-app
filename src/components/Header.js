import React from "react";
import { NavLink } from "react-router-dom";
import ExpenseTotal from "./ExpenseTotal";

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <ExpenseTotal />
    <ul>
      <li>
        <NavLink to="/" activeClassName="is-active" exact={true}>
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink to="/create" activeClassName="is-active">
          Add expanse
        </NavLink>
      </li>
      {/* <li>
        <NavLink to="/edit/:id" activeClassName="is-active">
          Edit expanse
        </NavLink>
      </li> */}
      <li>
        <NavLink to="/help" activeClassName="is-active">
          Help
        </NavLink>
      </li>
    </ul>
  </header>
);

export default Header;
