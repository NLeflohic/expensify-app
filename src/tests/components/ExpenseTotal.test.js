import React from "react";
import { shallow } from "enzyme";
import { ExpenseTotal } from "../../components/ExpenseTotal";

test("Should render ExpenseTotal correctly without expenses", () => {
  const wrapper = shallow(<ExpenseTotal expensesCount={0} />);
  expect(wrapper).toMatchSnapshot();
});

test("Should render ExpenseTotal correctly without expenses", () => {
  const wrapper = shallow(<ExpenseTotal expensesCount={1} expensesSum={444} />);
  expect(wrapper).toMatchSnapshot();
});
