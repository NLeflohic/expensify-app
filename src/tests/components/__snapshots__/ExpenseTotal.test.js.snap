// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseTotal correctly without expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
        0.00
         €
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render ExpenseTotal correctly without expenses 2`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
        0.00
         €
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;
